client/NetPacksClient.cpp

```diff
@@ -228,6 +228,7 @@ void ApplyClientNetPackVisitor::visitSetStackType(SetStackType & pack)
 void ApplyClientNetPackVisitor::visitEraseStack(EraseStack & pack)
 {
 	dispatchGarrisonChange(cl, pack.army, ObjectInstanceID());
+    cl.invalidatePaths(); //it is possible to remove last non-native unit for current terrain and lose movement penalty
 }
 
 void ApplyClientNetPackVisitor::visitSwapStacks(SwapStacks & pack)
```

Comment:
visitSwapStacks and visitInsertNewStack don't need path invalidation?

Reply:
May need investigation, did not see them triggering for some use cases I tested yesterday.